Use an explicit test matrix for Travis to get Python version

jdufresne opened the pull request:
Not all python versions are available in the default testing image. Instruct Travis to make the necessary Python version available in the test matrix.

Fixes Travis test failures.

gsnedders commented on .travis.yml:
```diff
@@ -1,26 +1,23 @@
-      env:
-        - TOXENV=py36
+      env: TOXENV=py36
+    - python: pypy
```
`pypy` is 2.5.0; we should probably just use `pypy-5.9` (i.e., the latest; sadly there's no alias for the latest current release).

jdufresne committed:
Use an explicit test matrix for Travis to get Python version

Not all python versions are available in the default testing image.
Instruct Travis to make the necessary Python version available in the
test matrix.

Fixes Travis test failures.

jdufresne commented:
The output of the PyPy build says:

```
$ python --version
Python 2.7.13 (c925e7381036, Jun 05 2017, 21:20:51)
[PyPy 5.8.0 with GCC 6.2.0 20160901]
```

So it looks like they have upgraded. When experimenting with `pypy-5.9`, Travis was unable to find a suitable image.

gsnedders commented:
OK, guess that changed despite what I was told before! (Surprised `pypy-5.9` doesn't work; implies they aren't running always-up-to-date python-build as I thought they were.)
